database/dbServices/shinx.api.js

```diff
@@ -40,14 +40,20 @@ module.exports = {
     },
     async switchShininessAndGet(id) {
         let shinx = await this.getShinx(id, ['user_id', 'shiny']);
-        return shinx.switchShininessAndGet();
+        let user = await this.getUser(id, ['user_id']);
+        let hasShinyCharm = await this.hasEventTrophy(user.user_id, 'Shiny Charm');
+        if (hasShinyCharm) {
+            return shinx.switchShininessAndGet();
+        } else {
+            throw new Error('User does not have the Shiny Charm');
+        }
```

Comment:
From what I've gathered this bug is not in the functionality for switching between shiny and non-shiny.
This seems to work correctly already.
The bug seems to be in users not receiving a shiny charm, or any level-up rewards, when reaching the required level.

Reply:
i will look into it tomorrow if i remember